modules/redpanda/redpanda_test.go

```diff
@@ -282,6 +282,50 @@ func TestRedpandaWithTLS(t *testing.T) {
 	require.Error(t, results.FirstErr(), kerr.UnknownTopicOrPartition)
 }
 
+func TestRedpandaWithTLSAndSASL(t *testing.T) {
+	cert, err := tls.X509KeyPair(localhostCert, localhostKey)
+	require.NoError(t, err, "failed to load key pair")
+
+	ctx := context.Background()
+
+	container, err := RunContainer(ctx,
+		WithTLS(localhostCert, localhostKey),
+		WithEnableSASL(),
+		WithEnableKafkaAuthorization(),
+		WithNewServiceAccount("superuser-1", "test"),
+		WithSuperusers("superuser-1"),
+	)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	})
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(localhostCert)
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      caCertPool,
+	}
+
+	broker, err := container.KafkaSeedBroker(ctx)
+	require.NoError(t, err)
+
+	kafkaCl, err := kgo.NewClient(
+		kgo.SeedBrokers(broker),
+		kgo.DialTLSConfig(tlsConfig),
+		kgo.SASL(scram.Auth{
+			User: "superuser-1",
+			Pass: "test",
+		}.AsSha256Mechanism()),
+	)
```

Comment:
I think it's fine with the current impl. I'd love to see anybody from Redpanda to take a look (maybe @gene-redpanda?).

Will wait a day for that, otherwise I'll merge this as is. Thanks!

Reply:
The client creation seems good. Maybe we can add at least one operation with the client, similar to what you do at https://github.com/testcontainers/testcontainers-go/pull/2048/files

Perhaps we can test listing topics and checking for no errors:

```
kafkaAdmCl := kadm.NewClient(kafkaCl)
_, err := kafkaAdmCl.ListTopics(ctx)
require.NoError(t, err)
```

Reply:
@srenatus would you be able to send a follow-up with what @r-vasquez suggested? It will come with lot of value for documentation purposes to have that as a testable example. Wdyt?

Reply:
https://github.com/testcontainers/testcontainers-go/pull/2200